test/common/jwt_authn/jwks_test.cc

```diff
@@ -0,0 +1,218 @@
+#include "common/jwt_authn/jwks.h"
+
+#include "test/test_common/utility.h"
+
+namespace Envoy {
+namespace JwtAuthn {
+namespace {
+
+TEST(JwksParseTest, GoodPem) {
+  // Public key PEM
+  const std::string jwks_text = "MIIBCgKCAQEAtw7MNxUTxmzWROCD5BqJxmzT7xqc9KsnAjbXCoqEEHDx4WBlfcwk"
+                                "XHt9e/2+Uwi3Arz3FOMNKwGGlbr7clBY3utsjUs8BTF0kO/poAmSTdSuGeh2mSbc"
+                                "VHvmQ7X/kichWwx5Qj0Xj4REU3Gixu1gQIr3GATPAIULo5lj/ebOGAa+l0wIG80N"
+                                "zz1pBtTIUx68xs5ZGe7cIJ7E8n4pMX10eeuh36h+aossePeuHulYmjr4N0/1jG7a"
+                                "+hHYL6nqwOR3ej0VqCTLS0OloC0LuCpLV7CnSpwbp2Qg/c+MDzQ0TH8g8drIzR5h"
+                                "Fe9a3NlNRMXgUU5RqbLnR9zfXr7b9oEszQIDAQAB";
+
+  auto jwks = Jwks::createFrom(jwks_text, Jwks::PEM);
+  EXPECT_EQ(jwks->getStatus(), Status::Ok);
+  EXPECT_EQ(jwks->keys().size(), 1);
+  EXPECT_TRUE(jwks->keys()[0]->pem_format_);
+}
+
+TEST(JwksParseTest, EmptyPem) {
+  auto jwks = Jwks::createFrom("", Jwks::PEM);
+  EXPECT_EQ(jwks->getStatus(), Status::JwksPemBadBase64);
+}
+
+TEST(JwksParseTest, BadPem) {
+  // U2lnbmF0dXJl is Base64 of "Signature"
+  auto jwks = Jwks::createFrom("U2lnbmF0dXJl", Jwks::PEM);
+  EXPECT_EQ(jwks->getStatus(), Status::JwksPemParseError);
+}
+
+TEST(JwksParseTest, GoodJwks) {
+  const std::string jwks_text =
```

Comment:
Use `R"( )"` literals so you don't have to escape `"`s and clearer to read (throughout)

Reply:
Done